File: packages/core/src/sentry.ts
// Client-side Sentry (PHILOSOPHY Observability / Cross-cutting): the SDK is
// @sentry/react-native — NOT the deprecated sentry-expo. Runtime init lives here;
// the BUILD-time half (production source maps / native symbolication) is the
// `@sentry/react-native/expo` config plugin + `getSentryExpoConfig` Metro wiring
// in each product's app (see app.config.ts / metro.config.js).
import * as Sentry from "@sentry/react-native";

import { env } from "./env";

export function initSentry(): void {
  if (!env.SENTRY_DSN) return; // no-op without a DSN (local dev, CI)
  Sentry.init({
    dsn: env.SENTRY_DSN,
    environment: env.ENV, // development | staging | production
    tracesSampleRate: 0.1,
  });
}

/** Tag the client-side scope with the per-request id the api wrapper generated —
 * matches the API middleware's tag, so one id links client event → API event → logs. */
export function captureRequestId(requestId: string): void {
  Sentry.setTag("request_id", requestId);
}

File: products/demo/api/src/demo_api/sentry.py
"""Server-side Sentry init (PHILOSOPHY Observability). No-op without a DSN,
so local dev and CI need no Sentry account. The middleware tags every event
with the request_id — the same tag the client-side SDK sets, which is what
makes client→API→logs traceability work.
"""

import sentry_sdk
from sentry_sdk.integrations.fastapi import FastApiIntegration
from sentry_sdk.integrations.starlette import StarletteIntegration

from .settings import get_settings


def init_sentry() -> None:
    settings = get_settings()
    if not settings.sentry_dsn:
        return  # no-op locally / in CI without a DSN
    sentry_sdk.init(
        dsn=settings.sentry_dsn,
        environment=settings.environment,
        release=settings.release,
        integrations=[FastApiIntegration(), StarletteIntegration()],
        traces_sample_rate=0.1,
        send_default_pii=False,
    )
